app/models/user.rb

```diff
@@ -1,6 +1,45 @@
 class User < ApplicationRecord
+  has_secure_password
+
+  before_save :downcase_email
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  MAX_LENGTH_NAME = 50
+  MAX_LENGTH_EMAIL = 255
+  MAX_AGE_YEARS = 100
+
+  validates :name, presence: true, length: {maximum: MAX_LENGTH_NAME}
+  validates :email, presence: true,
+                  length: {maximum: MAX_LENGTH_EMAIL},
+                  format: {with: VALID_EMAIL_REGEX},
+                  uniqueness: {case_sensitive: false}
+  validates :birthday, presence: true
+  validate :birthday_within_range
+
   has_many :microposts, dependent: :destroy
 
-  validates :name, presence: true
-  validates :email, presence: true, uniqueness: true
+  private
+
+  def downcase_email
+    email.downcase!
+  end
+
+  def birthday_within_range
+    return unless birthday
+
+    if birthday > Date.current
+      errors.add(:birthday,
+                 I18n.t("activerecord.errors.models.user.
+                 attributes.birthday.future_date"))
+    end
```

Comment:
```suggestion
    errors.add(:birthday, :future_date)) if birthday > Date.current
```
Nếu đã defind đúng cấu trúc messages lỗi rồi, thì có thể dùng tên key để trỏ đến messages luôn e nhé
Fix tương tự cho những chỗ khác